### DISCUSSION TITLE
`window` is not defined: web worker fix missing from v4.0?

### DISCUSSION THREAD

**Discussion by nrakochy** [vitejs/vite-plugin-react#292] [Q&A]
Hi there, 

@ArnaudBarre thanks for your work on this library. Rather than a new issue I thought I might open a discussion. Did #181 make it into the `v4` release? 

I just ran into this issue using Storybook with an `offscreencanvas` using a dedicated worker. It is especially difficult to troubleshoot, as I had to remove all the imports before I could see in the debugger that it was `vite` actually throwing the error rather than local code depending on `window`.

**Comment by ArnaudBarre** [comment]
This was included in the v4.0.1: https://github.com/vitejs/vite-plugin-react/releases/tag/plugin-react%404.0.1

If you're using at least this version, maybe the check is not working, this was kind of a hacky fix.

I hope some people will take care to figure out HMR for web workers before having full apps render ouside the dom, otherwise this will be a great regression in DX. Currently, if your code mostly uses workers, you get a simpler DX by removing the react plugin and having raw Vite esbuild + full reload.

**Comment by nrakochy** [comment]
Thanks for the quick response 🙏  It looks like Storybook is on [v3](https://github.com/storybookjs/storybook/blob/6b22f0e01cf979aa66ae26ba77dd5edc21a5947f/code/frameworks/react-vite/package.json#L54C1-L54C38). Apologies for the noise.
